hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -389,8 +399,9 @@ class AzureStorageFS(val credentialsJSON: Option[String] = None) extends FS {
     val prefixMatches = blobContainerClient.listBlobsByHierarchy(prefix)
 
     prefixMatches.forEach(blobItem => {
-      statList += fileStatus(url.withPath(blobItem.getName))
+      statList += AzureStorageFileStatus(blobItem)
```

Comment:
Ah, I think this is actually a bit of a bug. The name of a blobItem is like `/foo/bar/baz` but we need the URL which is `hail-az://account/container/foo/bar/baz`.